src/gitService.ts

```diff
@@ -925,13 +925,17 @@ export class GitService extends Disposable {
     }
 
     // TODO: Clear cache when git config changes
-    private _userNameMapCache: Map<string, string | undefined> = new Map();
+    private _userMapCache: Map<string, { name: string, email: string } | undefined> = new Map();
```

Comment:
I think this should be adapted like so
```ts
    private _userMapCache = new Map<string, { name?: string; email?: string } | null>();

    async getCurrentUser(repoPath: string) {
        let user = this._userMapCache.get(repoPath);
        if (user != null) return user;
        // If we found the repo, but no user data was found just return
        if (user === null) return undefined;

        const data = await Git.config_getRegex('user.(name|email)', repoPath);
        if (!data) {
            // If we found no user data, mark it so we won't bother trying again
            this._userMapCache.set(repoPath, null);
            return undefined;
        }

        user = { name: undefined, email: undefined };

        let match: RegExpExecArray | null = null;
        do {
            match = userConfigRegex.exec(data);
            if (match == null) break;

            // Stops excessive memory usage -- https://bugs.chromium.org/p/v8/issues/detail?id=2869
            user[match[1] as 'name' | 'email'] = (' ' + match[2]).substr(1);
        } while (match != null);

        this._userMapCache.set(repoPath, user);
        return user;
    }
```

Allows user name or email to be missing, and avoids calling config each time if we don't find data. It also allows for the last entry to win (since that is "closest" to the repo)

Reply:
The regex
```ts
const userConfigRegex = /^user\.(name|email) (.*)$/gm;
```

Can be placed on line 70 in `gitService.ts`